meson.build

```diff
@@ -286,21 +287,21 @@ if not get_option('tensorrt-support').disabled()
     cuda_dep = dependency('cuda-' + ver, required: false)
     cudart_dep = dependency('cudart-' + ver, required: false)
     if cuda_dep.found() and cudart_dep.found()
-      if ver != '11.0' and ver != '10.2'
+      if ver != '12.1' and ver != '11.0' and ver != '10.2'
         message('Warning: the recommended cuda version is at least 10.2')
       endif
       break
     endif
   endforeach
 
   nvinfer_lib = cxx.find_library('nvinfer', required: false)
-  if nvinfer_lib.found() and cxx.check_header('NvInfer.h')
+  if nvinfer_lib.found() and cxx.has_header('NvInfer.h')
     nvinfer_dep = declare_dependency(dependencies: nvinfer_lib)
   endif
 
-  nvparsers_lib = cxx.find_library('nvparsers', required: false)
-  if nvparsers_lib.found() and cxx.check_header('NvUffParser.h')
-    nvparsers_dep = declare_dependency(dependencies: nvparsers_lib)
+  nvonnxparser_lib = cxx.find_library('nvonnxparser', required: false)
+  if nvonnxparser_lib.found() and cxx.has_header('NvOnnxParser.h')
+    nvonnxparser_dep = declare_dependency(dependencies: nvonnxparser_lib)
```

Comment:
NvInferVersion.h contains the version of tensorrt:

```
...
#define NV_TENSORRT_MAJOR 10 //!< TensorRT major version.
...
```
Perhaps I should use that to determine the version of tensorrt?

Reply:
Done, see most recent commit.